relayer/main/main.go

```diff
@@ -176,7 +178,10 @@ func main() {
 		logger.Fatal("Failed to create app request network", zap.Error(err))
 		os.Exit(1)
 	}
-	defer network.Shutdown()
+	defer func() {
+		cancel()
+		network.Shutdown()
```

Comment:
If we want to wait for individual portions of the code to shutdown, I think we should propogate a `waitgroup` to them. That way we can wait on the waitgroup for all the services to exit, and we don't need odd things like this